src/main/java/sx/blah/discord/util/LongMapCollector.java

```diff
@@ -0,0 +1,56 @@
+package sx.blah.discord.util;
+
+import sx.blah.discord.handle.obj.IIDLinkedObject;
+import sx.blah.discord.util.cache.LongMap;
+
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+
+public class LongMapCollector<T extends IIDLinkedObject> implements Collector<T, List<T>, LongMap<T>> {
```

Comment:
Does koboloke not have this sort of utility built-in? Also this should be documented.